src/include/miopen/find_solution.hpp

```diff
@@ -431,11 +431,14 @@ struct SolverContainer
     {
         const auto network_config = problem.MakeNetworkConfig();
 
-        if(const auto existingInvoker =
-               ctx.GetStream().GetInvoker(network_config, boost::none, algo))
+        if(IsAnySolverApplicable(ctx, problem))
         {
-            (*existingInvoker)(ctx.GetStream(), invoke_params);
-            return;
+            if(const auto existingInvoker =
+                   ctx.GetStream().GetInvoker(network_config, boost::none, algo))
+            {
+                (*existingInvoker)(ctx.GetStream(), invoke_params);
+                return;
+            }
```

Comment:
@seungmanhan This looks like an optimization, as if the idea is that if all solvers are applicable, then we can take any existing invoker and use it (without constructing a new one).

However IsAnySolverApplicable does not check if all solvers are applicable. It checks if _at least one_ solver is applicable.

🔴 If I am right, then this optimization should be removed. Yeah, it may work, but only while we have only one solver (or, if we have several solvers, their applicability scopes do not intersect). But in the latter case the Invoker Cache will do the job of re-using the existing invoker.

Anyway, this optimization looks _very questionable_ (let me omit the details or now).

[Notice] ⚠️ Similar optimizations, if exist in other primitives, should be removed as well.

[Notice] We need to rename IsAnySolverApplicable to IsAtLeastOneSolverApplicable.

/CC @DrizztDoUrden @CAHEK7

Reply:
In the case of network config, it was created to be executed immediately when there is no problem even if the same kernel is run. However, because the current method passes invoke_params directly without checking isapplicable, it is executed even if isapplicable is false.
So, things related to isapplicable are added to the network config, which diminishes the advantage of using the network config.
To solve this, there is a way to check isapplicable in the constructor, but I thought it was not the right direction because it requires throwing.
So, I did SearchForSolutions and first checked whether slns was empty and modified it to find if it had the same network config. Please check it out.